src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsScheduling/Schedule.tsx

```diff
@@ -0,0 +1,189 @@
+      });
+  };
+
+  const onCancel = () => {
+    setWasTouched(false);
+    setCronString(schedule || defaultCronString);
+  };
+
+  const onCronInput = (value: string, position: number) => {
+    const split = cronString.split(' ');
+    setCronString([...split.slice(0, position), value, ...split.slice(position + 1)].join(' '));
+    setWasTouched(true);
+  };
+
+  const getNextExecution = () => {
+    try {
+      const nextExecution = parser.parseExpression(cronString).next().toDate();
+      const duration = intervalToDuration({start: new Date(), end: nextExecution});
+
+      if (!isValidFormat) {
+        setIsValidFormat(true);
+      }
+      return `in ${formatDuration(duration)}`;
+    } catch (e) {
+      if (isValidFormat) {
+        setIsValidFormat(false);
+      }
+      return 'Invalid cron format';
+    }
+  };
+
+  const [minute, hour, day, month, dayOfWeek] = cronString.split(' ');
+
+  return (
+    <ConfigurationCard
+      title="Schedule"
+      description={`You can add a cronjob like schedule for your ${namingMap[entity]} which will then be executed automatically.`}
+      onConfirm={onSave}
+      onCancel={onCancel}
+      isButtonsDisabled={!wasTouched}
+    >
+      <StyledSpace direction="vertical" size={32}>
+        <Select
+          placeholder="Quick select a schedule template"
+          style={{width: '100%'}}
+          options={quickOptions}
+          onSelect={(value: string) => {
+            setCronString(value);
+            setWasTouched(true);
+          }}
+        />
+        <StyledColumn>
+          <Text className="middle regular">Cron Format</Text>
+          <StyledCronFormat>
+            <CronInput title="Minute" value={minute} onChange={value => onCronInput(value, 0)} />
+            <CronInput title="Hour" value={hour} onChange={value => onCronInput(value, 1)} />
+            <CronInput title="Day" value={day} onChange={value => onCronInput(value, 2)} />
+            <CronInput title="Month" value={month} onChange={value => onCronInput(value, 3)} />
+            <CronInput title="Day / Week" value={dayOfWeek} onChange={value => onCronInput(value, 4)} />
+          </StyledCronFormat>
+        </StyledColumn>
+        <StyledRow>
+          <StyledColumn>
+            <Text className="middle regular" style={{color: isValidFormat ? Colors.slate200 : Colors.amber400}}>
+              Cron Preview{' '}
+              {!isValidFormat ? (
+                <Tooltip title="Cron input is invalid">
+                  <WarningOutlined />
+                </Tooltip>
+              ) : null}
+            </Text>
+            <Text style={{fontFamily: Fonts.robotoMono, color: Colors.slate400}} className="middle regular">
+              {cronString}
+            </Text>
+          </StyledColumn>
+          <StyledColumn>
+            <Text className="middle regular">Next Execution</Text>
+            <Text style={{color: Colors.slate400}} className="middle regular">
+              {getNextExecution()}
```

Comment:
Maybe it's worth to extract "duration" to a separate component, and make it update each second by itself